### DISCUSSION TITLE
WebFollower no data

### DISCUSSION THREAD

**Discussion by MitjaRazpet** [NightscoutFoundation/xDrip#3710] [Bug reports] [answered]
Hi. 
I have a problem with web follower that used to work normally and at some point just stopped working. I tried every instructions that i found but could not get it to work. My version of xdrio is dfcbe80-2024.09.17. I added debug logs for WebFollower and WebFollow and get this logs:
1 ^ 2024-10-10 22:55:31 ^ WebFollow ^ Anticipate next: 2024-10-10 23:00:30  last: 1970-01-01 01:00:00
1 ^ 2024-10-10 22:55:31 ^ WebFollow ^ FAILED AT STEP: aUrl|jput^email-U,password-P|country-data.user.country;token-data.authTicket.token|status-AZ
1 ^ 2024-10-10 22:55:31 ^ WebFollow ^ Pprocess exception: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.length()' on a null object reference
1 ^ 2024-10-10 22:55:31 ^ WebFollow ^ PProcessing psa: status -> AZ
1 ^ 2024-10-10 22:55:31 ^ WebFollower ^ RES: {"status":0,"data":{"redirect":true,"region":"eu"}}

1 ^ 2024-10-10 22:55:30 ^ WebFollower ^ REQUEST URL: https://api.libreview.io/llu/auth/login
1 ^ 2024-10-10 22:55:30 ^ WebFollow ^ Processing step: aUrl|jput^email-U,password-P|country-data.user.country;token-data.authTicket.token|status-AZ
1 ^ 2024-10-10 22:55:30 ^ WebFollow ^ Processing step: tSet:skip3|
1 ^ 2024-10-10 22:55:30 ^ WebFollow ^ No last reading
1 ^ 2024-10-10 22:55:30 ^ WebFollow ^ Processing step: allCurrent:quit|
1 ^ 2024-10-10 22:55:30 ^ WebFollow ^ Anticipate next: 2024-10-10 23:00:30  last: 1970-01-01 01:00:00
1 ^ 2024-10-10 22:55:30 ^ WebFollow ^ WAKE UP WAKE UP WAKE UP

**Comment by Navid200** [comment]
Have you seen this?  

https://navid200.github.io/xDrip/docs/Libre3_Follow.html

**Reply by PhilTU** [reply]
Are you in the EU ?

**Reply by MitjaRazpet** [reply]
Yes.Im from Slovenia, Europe. Maybe there is something with redirecting to another coutry as messege sais reddirect: true?

**Reply by PhilTU** [reply]
there are servers around the world, I think it's just redirecting you to the appropriate one.

Is data received in LibreLinkUp app with same credentials ?

**Reply by MitjaRazpet** [reply]
Yes it works in LibreLinkUp with no problem, same credentials. Xdrip worked till about a week ago and then just stopped recieving data.
Wierd is that after redirect there is no other request? Status 0 means ok?

**Reply by MitjaRazpet** [reply]
now I tried to install xdrip on another phone and i get the same results.

**Reply by PhilTU** [reply]
I have had web follower fail completely too. It isn't easy to diagnose.

**Reply by MitjaRazpet** [reply, marked as answer]
Now I managed to get it to work. I opened a VPN connection from Slovenia to Slovenia and the next pool was successful. After that i closed the VPN and it works normally. Dont know why this helped, but if someone have the same problem try to open vpn to your country and wait for the app to read the data once. After that you can close the vpn.
Thank PhilTU for your help.

**Comment by MitjaRazpet** [comment]
Yes I have tried all these things but still not working. I am using Libre 2, but this is probably the same. I tried all this once again but still no luck. I also tried to create new email for librelinkup, and configured as in instructions but still it doesnt work. If i change the password to wrong one i get this error 1 ^ 2024-10-11 05:35:38 ^ WebFollower ^ RES: {"status":2,"error":{"message":"notAuthenticated"}}
